libres/lib/job_queue/lsf_driver.cpp

```diff
@@ -959,24 +927,61 @@ void lsf_driver_free_job(void *__job) {
     lsf_job_free(job);
 }
 
+namespace lsf {
+std::vector<std::string> split_string(std::string str, std::string delimiter) {
+    std::vector<std::string> splitted = {};
+    size_t pos = 0, next_pos = 0;
+
+    while ((next_pos = str.find(delimiter, pos)) != std::string::npos) {
+        splitted.push_back(str.substr(pos, next_pos - pos));
+        pos = next_pos + delimiter.size();
+    }
+    splitted.push_back(str.substr(pos, str.size() - pos));
+    return splitted;
+}
+
+/*
+ * Parses the given file containing colon separated hostnames, ie.
+ * "hname1:hname2:hname3". This is the same format as
+ * written by lsf_job_write_bjobs_to_file().
+ */
+std::vector<std::string> parse_hostnames(const char *fname) {
+    std::ifstream stream(fname);
+
+    std::string line;
+    if (std::getline(stream, line)) { //just read the first line
+        // bjobs uses : as delimiter
+        std::vector<std::string> hosts_w_star = split_string(line, ":");
+
+        // change "4*be-lsf01" to "be-lsf01"
+        std::vector<std::string> hosts;
+        std::transform(hosts_w_star.begin(), hosts_w_star.end(),
+                       std::back_inserter(hosts),
+                       [](const std::string &host) -> std::string {
+                           return split_string(host, "*").back();
+                       });
+
+        return hosts;
+    }
+    return {};
+}
+} // namespace lsf
+
 static void lsf_driver_node_failure(lsf_driver_type *driver,
                                     const lsf_job_type *job) {
     long lsf_job_id = lsf_job_get_jobnr(job);
     char *fname =
         lsf_job_write_bjobs_to_file(driver->bjobs_cmd, driver, lsf_job_id);
-    stringlist_type *hosts = lsf_job_alloc_parse_hostnames(fname);
-    char *hostnames = stringlist_alloc_joined_string(hosts, ", ");
+    auto hostnames = join_strings(lsf::parse_hostnames(fname), ",");
```

Comment:
Where does `join_strings` come from?

Reply:
This is a bit of s contention point as there are some existing implementations of split/join strings in the driver implementations. Here an existing implementation in the lsf driver is reused, but perhaps some effort should be out into having just one implementation. I gave that an attempt, however, these implementations have some diverging behavior so I didn't finish that work.

Reply:
Moved `join_strings` into the `ert/res_util/string.hpp`